### PR TITLE
Improve data parsing errors in runtime hosted parsing functions

### PR DESCRIPTION
**Pull request by fordN**
The hosted function exports `bytes_to_string()` and `json_from_bytes()` have been updated to return helpful error messages.  Since these parsing functions are called from  a subgraph mapping, context on the `graph-ts` function used can be essential for efficient debugging. 

Related to the `graph-ts` issue to [make toHex() more discoverable](https://github.com/graphprotocol/graph-ts/issues/41).

### CODE REVIEW

**Comment by fordN** [comment]
Dependent on [graph-ts pull#43](https://github.com/graphprotocol/graph-ts/pull/43) because it mentions `toHexString()`.

File: runtime/wasm/src/host_exports.rs [hunk 1]
```diff
@@ -246,7 +246,15 @@ where
         &self,
         bytes: Vec<u8>,
     ) -> Result<String, HostExportError<impl ExportError>> {
-        let s = String::from_utf8(bytes).map_err(HostExportError)?;
+        let s = String::from_utf8(bytes).map_err(|e| {
+            HostExportError(format!(
+                "Failed to parse Byte array using `toString()`. This may be caused by attempting \
```
Comment (Jannis) [inline runtime/wasm/src/host_exports.rs]: `Byte array` -> `byte array`

File: runtime/wasm/src/host_exports.rs [hunk 2]
```diff
@@ -246,7 +246,15 @@ where
...
+                 to convert a value such as an address that cannot be parsed to a unicode string. \
+                 Try 'toHexString()' instead. error: {error},  bytes:`{bytes:?}`",
+                error = e.utf8_error(),
```
Comment (Jannis) [inline runtime/wasm/src/host_exports.rs]: Named arguments to `format!` are a thing? Nice!

File: runtime/wasm/src/host_exports.rs [hunk 3]
```diff
@@ -246,7 +246,15 @@ where
         &self,
         bytes: Vec<u8>,
     ) -> Result<String, HostExportError<impl ExportError>> {
-        let s = String::from_utf8(bytes).map_err(HostExportError)?;
+        let s = String::from_utf8(bytes).map_err(|e| {
+            HostExportError(format!(
+                "Failed to parse Byte array using `toString()`. This may be caused by attempting \
+                 to convert a value such as an address that cannot be parsed to a unicode string. \
+                 Try 'toHexString()' instead. error: {error},  bytes:`{bytes:?}`",
```
Comment (Jannis) [inline runtime/wasm/src/host_exports.rs]: I'd turn this around: ```Bytes: `{bytes:?}`. Error: {error}```

File: runtime/wasm/src/host_exports.rs [hunk 4]
```diff
@@ -303,7 +311,13 @@ where
         &self,
         bytes: Vec<u8>,
     ) -> Result<serde_json::Value, HostExportError<impl ExportError>> {
-        serde_json::from_reader(&*bytes).map_err(HostExportError)
+        serde_json::from_reader(&*bytes).map_err(|e| {
+            HostExportError(format!(
+                "Failed to parse byte array to JSON. error: {error}, bytes: `{bytes:?}`",
```
Comment (Jannis) [inline runtime/wasm/src/host_exports.rs]: Similar here. I'd do ```Failed to parse JSON from byte array: Bytes: `{bytes:?}`. Error: {error}```

File: runtime/wasm/src/host_exports.rs [hunk 5]
```diff
@@ -246,7 +246,15 @@ where
         &self,
         bytes: Vec<u8>,
     ) -> Result<String, HostExportError<impl ExportError>> {
-        let s = String::from_utf8(bytes).map_err(HostExportError)?;
+        let s = String::from_utf8(bytes).map_err(|e| {
+            HostExportError(format!(
+                "Failed to parse byte array using `toString()`. This may be caused by attempting \
+                 to convert a value such as an address that cannot be parsed to a unicode string. \
+                 Try 'toHexString()' instead. Bytes:`{bytes:?}`. Error: {error}",
```
Comment (Jannis) [inline runtime/wasm/src/host_exports.rs]: There's a space missing after `Bytes:`.

Comment (fordN) [reply runtime/wasm/src/host_exports.rs]: @Jannis Updated! Thanks for noticing the spacing. 🥇

**Pull request merged**